src/settings/controllers/SlidingSyncController.ts

```diff
@@ -38,6 +92,14 @@ export default class SlidingSyncController extends SettingController {
         if (SettingsStore.getValue("feature_sliding_sync")) {
             return _t("labs|sliding_sync_disabled_notice");
         }
+        // Server doesn't support sliding sync
+        this.slidingSyncHealthCheck().then(
+            (supportsSlidingSync) => {
+                if (!supportsSlidingSync) {
+                    return _t("labs|sliding_sync_server_no_support");
+                }
+            }
+        );
```

Comment:
This loses the race to the `return` below. I don't know how to implement this bit, advice appreciated!

Reply:
```
        const supportsSlidingSync = await this.slidingSyncHealthCheck();
		if (!supportsSlidingSync) {
			return ...
		}
```

Otherwise it isn't a race, as the return is in a completely different execution given you aren't awaiting the promise

Reply:
The problem is `settingDisabled()` isn't an async function so I can't use `await` (hence me trying and failing to use `then`!)

Reply:
Good point, you'd need to have some async code write to a static member on SlidingSyncController which stores the healthcheck

Reply:
Sorry, you're going to have to help me a little more on this one I'm afraid! Do you have an example to hand?

Reply:
Where you call `SlidingSyncManager.instance.setup` you could have an `else` clause which calls something like `SlidingSyncManager.instance.checkSupport` and that method awaits the wellKnown and does all the various support checks and then writes the result of those into a static member on the SlidingSyncController (e.g. SlidingSyncController.serverSupport = ...` which then the disabled method reads